secant/Features/App/AppDestination.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  AppDestination.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 01.12.2022.
+//
+
+import Foundation
+import ComposableArchitecture
+import ZcashLightClientKit
+
+/// In this file is a collcetion of helpers that control all state and action related operations
+/// for the `AppReducer` with a connection to the UI navigation.
+extension AppReducer {
+    struct DestinationState: Equatable {
+        enum Destination: Equatable {
+            case welcome
+            case startup
+            case onboarding
+            case sandbox
+            case home
+            case phraseValidation
+            case phraseDisplay
+        }
+        
+        var prevDestination: Destination?
+        var internalDestination: Destination = .welcome
+        
+        var destination: Destination {
+            get { internalDestination }
+            set {
+                prevDestination = internalDestination
+                internalDestination = newValue
+            }
+        }
+    }
+    
+    enum DestinationAction: Equatable {
+        case deeplink(URL)
+        case deeplinkHome
+        case deeplinkSend(Zatoshi, String, String)
+        case updateDestination(AppReducer.DestinationState.Destination)
+    }
+
+    // swiftlint:disable:next cyclomatic_complexity
+    func destinationReduce() -> Reduce<AppReducer.State, AppReducer.Action> {
+        Reduce { state, action in
+            switch action {
+            case let .destination(.updateDestination(destination)):
+                state.destinationState.destination = destination
+
+            case .sandbox(.reset):
+                state.destinationState.destination = .startup
+
+            case .phraseValidation(.proceedToHome):
+                state.destinationState.destination = .home
+
+            case .phraseValidation(.displayBackedUpPhrase):
+                state.destinationState.destination = .phraseDisplay
+
+            case .phraseDisplay(.finishedPressed):
+                // user is still supposed to do the backup phrase validation test
+                if state.destinationState.prevDestination == .welcome
+                || state.destinationState.prevDestination == .onboarding {
+                    state.destinationState.destination = .phraseValidation
+                }
+                // user wanted to see the backup phrase once again (at validation finished screen)
+                if state.destinationState.prevDestination == .phraseValidation {
+                    state.destinationState.destination = .home
+                }
+
+            case .destination(.deeplink(let url)):
+                // get the latest synchronizer state
+                var synchronizerStatus = SDKSynchronizerState.unknown
+                _ = sdkSynchronizer.stateChanged.sink { synchronizerStatus = $0 }
```

Comment:
This seems strange. How does this work? Can `stateChanged` emit value faster then next guard is executed? Wouldn't be something like `synchronizerStatus = sdkSynchronizer.stateChanged.value` better? The code doesn't even wait for this sequence to emit. I don't a reason to use react here.

Reply:
fixed